WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java

```diff
@@ -162,7 +162,14 @@ public static void viewCurrentSite(Context context, SiteModel site, boolean open
         AnalyticsTracker.Stat stat = openFromHeader ? AnalyticsTracker.Stat.OPENED_VIEW_SITE_FROM_HEADER
                 : AnalyticsTracker.Stat.OPENED_VIEW_SITE;
         AnalyticsUtils.trackWithSiteDetails(stat, site);
-        openUrlExternal(context, site.getUrl());
+
+        final String siteUrl = site.getUrl();
```

Comment:
Given the rarity of this bug, what do you think about doing away with the additional string resource and changing the check at the top of this method from `if (site == null)`  to `if (site == null || site.getUrl() == null)`?

Reply:
Fair point @nbradbury . I refactored the null checks a bit with 891c6d3. BTW, the Track now fires even on null SiteModel so we can have good Analytics of the user trying to view the site.

I removed the differentiated toast _to the user_ as you suggested. I kept the log message though as it's useful for debugging (think: when user contacts Support).